priority_queue.py

```diff
@@ -0,0 +1,57 @@
+class PNode(object):
+    def __init__(self, val=None, priority=0):
+        self.val = val
+        self.priority = priority
+
+
+class Priority_Queue(object):
```

Comment:
I'm finding it hard to believe that you repeated the entire code of the `bin_heap` from the previous assignment instead of just composing that class into this one. Imagine the possibilities:

```python
from binary_heap import Heap

class PriorityQueue(object):

    def __init__(self):
        self.heap = Heap()
```
What happens to the rest of your code if you have that at the start here?